src/main.rs

```diff
@@ -39,45 +43,36 @@ fn main() {
 
     // === Sign in, if user has a token ===
     if !(*auth_store).borrow().is_empty() {
-        fetch_data(&ui, &auth_store.clone());
+        let mut auth_store = (*auth_store).borrow_mut();
+        let mut auths_to_remove = vec![];
+        for (i, auth) in auth_store.iter_mut().enumerate() {
+            match fetch_data(auth.platform.clone(), auth.token.clone(), &ui) {
+                Ok(..) => (),
+                (Err(error)) if error.status() == StatusCode::Unauthorized => {
+                    eprintln!("Token expired");
+                    auths_to_remove.push(i);
+                }
+                _ => eprintln!("There has been an issue with internet connection"),
+            }
+        }
+        auths_to_remove.sort_by(|a, b| b.cmp(a));
+        auths_to_remove.iter().for_each(|i| auth_store.remove(*i));
```

Comment:
This feels like too much logic to be in the main, try moving in somewhere else. For example allow `fetch_data` to work with a slice of an array.